### PR TITLE
Add SSL options to producer

### PR DESCRIPTION
**Pull request by suyograo**
According to https://kafka.apache.org/documentation.html#security_configclients
SSL options are the same for producer and consumers so these options
are consistent with logstash-input-kafka

### CODE REVIEW

File: lib/logstash/outputs/kafka.rb [hunk 1]
```diff
@@ -156,6 +167,16 @@ def create_producer
       props.put(kafka::RETRY_BACKOFF_MS_CONFIG, retry_backoff_ms.to_s)
       props.put(kafka::SEND_BUFFER_CONFIG, send_buffer_bytes.to_s)
       props.put(kafka::VALUE_SERIALIZER_CLASS_CONFIG, value_serializer)
+      
+      if ssl
+        props.put("security.protocol", "SSL")
+        props.put("ssl.truststore.location", ssl_truststore_location)
```
Comment (talevy) [inline lib/logstash/outputs/kafka.rb]: why not

```
props.put("ssl.truststore.location", ssl_truststore_location) unless ssl_truststore_location.nil?
```

just like the rest

Comment (suyograo) [inline lib/logstash/outputs/kafka.rb]: I like the exception to bubble up if `ssl` and `ssl_truststore_location` is not set. That's the least required for SSL

Comment (talevy) [inline lib/logstash/outputs/kafka.rb]: does it not bork regardless?

Comment (suyograo) [inline lib/logstash/outputs/kafka.rb]: Nope, this way it says for SSL, you need `ssl_truststore_location` to be set (From the Kafka code)

**Comment by suyograo** [comment]
@talevy Changed to add config error

**Comment by talevy** [comment]
LGTM

**Pull request closed without merging**